src/cloud-cli/meltano/cloud/cli/project.py

```diff
@@ -0,0 +1,221 @@
+        project["git_repository"],
+    )
+
+
+def _format_projects_table(projects: list[CloudProject], table_format: str) -> str:
+    """Format the projects as a table.
+
+    Args:
+        projects: The projects to format.
+
+    Returns:
+        The formatted projects.
+    """
+    return tabulate.tabulate(
+        [_process_table_row(project) for project in projects],
+        headers=(
+            "Active",
+            "Name",
+            "Git Repository",
+        ),
+        tablefmt=table_format,
+        # To avoid a tabulate bug (IndexError), only set colalign if there are projects
+        colalign=("center", "left", "left") if projects else (),
+    )
+
+
+private_project_attributes = {"tenant_resource_key", "project_id"}
+
+
+def _remove_private_project_attributes(project: CloudProject) -> dict[str, t.Any]:
+    return {k: v for k, v in project.items() if k not in private_project_attributes}
+
+
+project_list_formatters = {
+    "json": lambda x: json.dumps(x, indent=2),
+    "markdown": lambda x: _format_projects_table(x, table_format="github"),
+    "terminal": lambda x: _format_projects_table(x, table_format="rounded_outline"),
+}
+
+
+@project_group.command("list")
+@click.option(
+    "--limit",
+    required=False,
+    type=int,
+    default=DEFAULT_GET_PROJECTS_LIMIT,
+    help="The maximum number of projects to display.",
+)
+@click.option(
+    "--format",
+    "output_format",
+    required=False,
+    default="terminal",
+    type=click.Choice(("terminal", "markdown", "json")),
+    help="The output format to use.",
+)
+@pass_context
+@run_async
+async def list_projects(
+    context: MeltanoCloudCLIContext,
+    output_format: str,
+    limit: int,
+) -> None:
+    """List Meltano Cloud projects."""
+    click.echo(
+        project_list_formatters[output_format](
+            [
+                _remove_private_project_attributes(x)
+                for x in (await _get_projects(config=context.config, limit=limit))
+            ],
+        ),
+    )
+
+
+async def _get_project_to_activate(
+    config: MeltanoCloudConfig,
+    identifier: str,
+) -> list[CloudProject]:
+    try:
+        ULID.from_str(identifier)
+    except ValueError:
+        return await _get_projects(config=config, project_name=identifier)
+    try:
+        return await _get_projects(config=config, project_id=identifier)
+    except MeltanoCloudError as ex:
+        if ex.response.status == HTTPStatus.NOT_FOUND:
+            # The provided identifier looks like an ID, but isn't one.
+            return await _get_projects(config=config, project_name=identifier)
+        raise
+
+
+@project_group.command("activate")
+@click.argument("identifier")
+@pass_context
+@run_async
+async def activate_project(
+    context: MeltanoCloudCLIContext,
+    identifier: str,
+) -> None:
+    """Set a project as active within the current Meltano Cloud session."""
+    projects = await _get_project_to_activate(context.config, identifier)
```

Comment:
Should we make the `identifier` arg optional and the give the user an interactive prompt to select a project if it's not set?

Reply:
@edgarrmondragon I agree that an interactive prompt would be nice, and it can be done using `@click.option(..., prompt=True)`. I'd like to combine this with `type=click.Choice`, but that presents 2 challenges:

- The interactive prompt for a `click.Choice` isn't great. By default it looks like this (for an option `--name` with 3 possible values `option 1`, `option 2` and `option 3`):
  ```
  Name (option 1, option 2, option 3): 
  ```
  I'd prefer it print the options as a numbered list, and then let the user enter the number corresponding to the option they'd like to select.
- `click.Choice` needs the option when `meltano.cloud.cli.project` is run, rather than when the `project activate` function is run. It would need to be extended to take a callback function that get the possible choices. Then that callback function could make the API call to get the available projects.

Reply:
Perhaps [`questionary`](https://github.com/tmbo/questionary) can help here, as detailed in this StackOverflow answer: https://stackoverflow.com/a/60425679/5946921

Reply:
Yeah, `questionary` looks like the Python equivalent of Golang's [Survey](https://github.com/go-survey/survey), which is used by `gh` and other CLIs.

I took a quick look at [Rich's support for prompts](https://rich.readthedocs.io/en/stable/prompt.html) and it's not much better than Click's.

Reply:
cc @kgpayne since he's worked with nice Rich prompts

Reply:
Implemented in 033c58f. This was somewhat complex, and difficult to test, but the user experience is vastly improved!